utils/utils.go

```diff
@@ -136,24 +138,40 @@ func CreateClient(conf NetConf) (*client.Client, error) {
 }
 
 func EnableDebugLogging() {
+	// Enable glogging for libcalico
 	_ = flag.Set("logtostderr", "true")
 	_ = flag.Set("v", "10")
 	_ = flag.Set("stderrthreshold", "10")
 	flag.Parse()
-	glog.Info("Calico CNI debug logging configured")
+	glog.Info("libcalico glog logging configured")
+
+	// Change logging level for CNI plugin
+	log.SetLevel(log.DebugLevel)
 }
 
 // ReleaseIPAM is called to cleanup IPAM allocations if something goes wrong during
 // CNI ADD execution.
 func ReleaseIPAllocation(ipamType string, stdinData []byte) {
-	glog.V(3).Infof("Cleaning up IP allocations for failed ADD")
+	log.Info("Cleaning up IP allocations for failed ADD")
 	if err := os.Setenv("CNI_COMMAND", "DEL"); err != nil {
 		// Failed to set CNI_COMMAND to DEL.
-		glog.Warningf("Failed to set CNI_COMMAND=DEL")
+		log.Warning("Failed to set CNI_COMMAND=DEL")
+	} else {
+		if err := ipam.ExecDel(ipamType, stdinData); err != nil {
+			// Failed to cleanup the IP allocation.
+			log.Warning("Failed to clean up IP allocations for failed ADD")
+		}
 	}
+}
 
-	if err := ipam.ExecDel(ipamType, stdinData); err != nil {
-		// Failed to cleanup the IP allocation.
-		glog.Warningf("Failed to clean up IP allocations for failed ADD")
-	}
+// Create a logger which always includes common fields
+func CreateContextLogger(args *skel.CmdArgs) *log.Entry {
+	// A common pattern is to re-use fields between logging statements by re-using
+	// the logrus.Entry returned from WithFields()
+	contextLogger := log.WithFields(log.Fields{
+		"ContainerID": args.ContainerID,
+		"NetNS":       args.Netns,
+	})
```

Comment:
Why don't we set the log level here from the value specified in the NetConf?  We already support a "log_level" field.  We can get rid of the "debug: true|false" one.